tests/13logout.pl

```diff
@@ -0,0 +1,48 @@
+test "Can logout current device",
+   requires => [ local_user_fixture( with_events => 0 ) ],
+
+   do => sub {
+      my ( $user ) = @_;
+
+      my $other_login;
+
+      matrix_login_again_with_user( $user )
+      ->then( sub {
+         ( $other_login ) = @_;
+
+         do_request_json_for( $user,
+            method => "POST",
+            uri    => "/r0/logout",
+            content => {},
+         )
+      })->then( sub {
+         matrix_sync( $user );
+      })->main::expect_http_401->then( sub {
```

Comment:
The `expect_http_401` should be attached to the sync request directly, so it only checks the result of that. The way it's written here it would catch (and consider as "OK") a 401 failure from any of the preceding logic; e..g the login or logout requests. This wants to be

    ->then( sub {
       matrix_sync( $user )->main::expect_http_401
    })->then( sub { ...